myuw/dao/mailman.py

```diff
@@ -0,0 +1,109 @@
+"""
+This class encapsulates the interactions with
+the uwnetid subscription resource.
+"""
+
+import logging
+import re
+from restclients.models.sws import Section, Term
+from myuw.dao.pws import get_person_of_current_user
+from restclients.mailman.list_checker import exists_section_list,\
+    exists_secondary_section_combined_list, get_section_list_name,\
+    exists_instructor_term_combined_list, get_instructor_term_list_name,\
+    get_secondary_section_combined_list_name
+from myuw.dao.instructor_schedule import _get_instructor_schedule
+
+
+logger = logging.getLogger(__name__)
+
+
+def _get_section(term, curriculum_abbr, course_number, section_id):
+    return Section(term=term,
+                   curriculum_abbr=curriculum_abbr,
+                   course_number=course_number,
+                   section_id=section_id)
+
+
+def get_single_email_list_by_section(section):
+    """
+    @return json of the section specfic email list info
+    """
+    return {
+        "section_id": section.section_id,
+        "list_address": get_section_list_name(section),
+        "list_exists": exists_section_list(section)
+        }
+
+
+def get_email_list(term, curriculum_abbr, course_number, section_id):
+    return get_single_email_list_by_section(
+        _get_section(term, curriculum_abbr, course_number, section_id))
+
+
+def get_email_lists_by_term(term):
+    """
+    @return json of the corresponding email lists
+    associated with the current instructor in the given term
+    """
+    person = get_person_of_current_user()
+    schedule = _get_instructor_schedule(person, term)
+    json_data = {
+        "year": term.year,
+        "quarter": term.quarter,
+        }
+    json_data["email_lists"] = []
+    for section in schedule.sections:
+        json_data["email_lists"].append(
```

Comment:
somewhat related, what happens for the users with very many sections or secondary sections?